libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1238,26 +1252,68 @@ blk_hash_t PbftManager::proposePbftBlock_() {
   // get DAG block and transaction order
   const auto propose_period = pbft_chain_->getPbftChainSize() + 1;
   auto dag_block_order = dag_mgr_->getDagBlockOrder(dag_block_hash, propose_period);
+
   if (dag_block_order.empty()) {
     LOG(log_er_) << "DAG anchor block hash " << dag_block_hash << " getDagBlockOrder failed in propose";
     assert(false);
   }
-  std::vector<trx_hash_t> non_finalized_transactions;
+
   std::unordered_set<trx_hash_t> trx_hashes_set;
   std::vector<trx_hash_t> trx_hashes;
+  u256 total_weight = 0;
+  uint32_t dag_blocks_included = 0;
   for (auto const &blk_hash : dag_block_order) {
     auto dag_blk = dag_blk_mgr_->getDagBlock(blk_hash);
     if (!dag_blk) {
       LOG(log_er_) << "DAG anchor block hash " << dag_block_hash << " getDagBlock failed in propose for block "
                    << blk_hash;
       assert(false);
     }
-    for (auto const &trx_hash : dag_blk->getTrxs()) {
+    u256 dag_block_weight = 0;
+    const auto &estimations = dag_blk->getEstimations();
```

Comment:
again, something like getTxsGasEstimations would make it much clearer of what you are actually getting here